openpaas-james/mailbox/encrypted/memory/src/main/java/com/linagora/openpaas/encrypted/EncryptedMailboxManager.scala

```diff
@@ -0,0 +1,123 @@
+package com.linagora.openpaas.encrypted
+
+import java.util.Optional
+import java.{lang, util}
+
+import javax.inject.Inject
+import org.apache.james.core.Username
+import org.apache.james.mailbox.MailboxManager.{MailboxCapabilities, MailboxRenamedResult, MessageCapabilities, SearchCapabilities}
+import org.apache.james.mailbox.model.search.MailboxQuery
+import org.apache.james.mailbox.model.{Mailbox, MailboxACL, MailboxAnnotation, MailboxAnnotationKey, MailboxId, MailboxMetaData, MailboxPath, MessageId, MessageRange, MultimailboxesSearchQuery}
+import org.apache.james.mailbox.{MailboxManager, MailboxSession, MessageManager}
+import org.reactivestreams.Publisher
+import reactor.core.publisher.Flux
+
+class EncryptedMailboxManager @Inject()(mailboxManager: MailboxManager) extends MailboxManager {
+
+  override def getSupportedMailboxCapabilities: util.EnumSet[MailboxCapabilities] = mailboxManager.getSupportedMailboxCapabilities
+
+  override def hasCapability(capability: MailboxCapabilities): Boolean = mailboxManager.hasCapability(capability)
+
+  override def getSupportedMessageCapabilities: util.EnumSet[MessageCapabilities] = mailboxManager.getSupportedMessageCapabilities
+
+  override def getSupportedSearchCapabilities: util.EnumSet[SearchCapabilities] = mailboxManager.getSupportedSearchCapabilities
+
+  override def getMailbox(mailboxPath: MailboxPath, session: MailboxSession): MessageManager = new EncryptedMessageManager(mailboxManager.getMailbox(mailboxPath, session), new InMemoryKeystoreManager())
```

Comment:
`new InMemoryKeystoreManager()` should be a constructor parameter (field)

We can care about injects later.